sdk/tables/azure-data-tables/tests/test_table_batch_cosmos.py

```diff
@@ -690,6 +690,18 @@ def test_batch_with_specialchar_partitionkey(self, tables_cosmos_account_name, t
         finally:
             self._tear_down()
 
+    @cosmos_decorator
+    @recorded_by_proxy
+    def test_empty_batch(self, tables_cosmos_account_name, tables_primary_cosmos_account_key):
+        url = self.account_url(tables_cosmos_account_name, "cosmos")
+        table_name = self.get_resource_name("mytable")
+        with TableClient(url, table_name, credential=tables_primary_cosmos_account_key) as client:
+            client.create_table()
+            with pytest.raises(HttpResponseError) as ex:
```

Comment:
That's interesting - so only Storage supports an empty batch - not Cosmos?

Reply:
There's an error in Storage with the same type as in Cosmos, the odd thing is the response in Storage is 202 which make it treated as `accepted`. For example, this is the internal_response in Storage:
```text
(Pdb) pp(vars(response.internal_response))
{'_content': b'--batchresponse_ad44525b-f096-44f6-b281-0ea47eee4461\r\nConten'
             b't-Type: multipart/mixed; boundary=changesetresponse_f6d0454a-5fa'
             b'0-42a0-bdb6-ca0447566474\r\n\r\n--changesetresponse_f6d0454a-5fa'
             b'0-42a0-bdb6-ca0447566474\r\nContent-Type: application/http\r\nCo'
             b'ntent-Transfer-Encoding: binary\r\n\r\nHTTP/1.1 400 Bad Requ'
             b'est\r\nDataServiceVersion: 1.0;\r\nContent-Type: application/xml'
             b';charset=utf-8\r\n\r\n<?xml version="1.0" encoding="utf-8"?><err'
             b'or xmlns="http://schemas.microsoft.com/ado/2007/08/dataservices/'
             b'metadata"><code>InvalidInput</code><message xml:lang="en-US">An '
             b'error occurred while processing this request.\nRequestId:b4ab'
             b'89fe-b002-0001-4768-c908ed000000\nTime:2023-08-07T19:50:44.01'
             b'67593Z</message></error>\r\n--changesetresponse_f6d0454a-5fa0-'
             b'42a0-bdb6-ca0447566474--\r\n--batchresponse_ad44525b-f096-44f6'
             b'-b281-0ea47eee4461--\r\n',
 '_content_consumed': True,
 '_next': None,
 'connection': <azure.core.pipeline.transport._bigger_block_size_http_adapters.BiggerBlockSizeHTTPAdapter object at 0x000001CCC05B17D0>,
 'cookies': <RequestsCookieJar[]>,
 'elapsed': datetime.timedelta(microseconds=29417),
 'encoding': None,
 'headers': {'Cache-Control': 'no-cache', 'Transfer-Encoding': 'chunked', 'Content-Type': 'multipart/mixed; boundary=batchresponse_ad44525b-f096-44f6-b281-0ea47eee4461', 'Server': 'Windows-Azure-Table/1.0 Microsoft-HTTPAPI/2.0', 'x-ms-request-id': 'b4ab89fe-b002-0001-4768-c908ed000000', 'x-ms-client-request-id': 'b2cd873d-355b-11ee-b74a-5cf37093a909', 'x-ms-version': '2019-02-02', 'X-Content-Type-Options': 'nosniff', 'Date': 'Mon, 07 Aug 2023 19:50:43 GMT'},
 'history': [],
 'raw': <urllib3.response.HTTPResponse object at 0x000001CCC05D8790>,
 'reason': 'Accepted',
 'request': <PreparedRequest [POST]>,
 'status_code': 202,
 'url': 'https://yalltbtestsprim.table.core.windows.net/$batch'}
```